pkgs/development/libraries/qt-5/5.15/default.nix

```diff
@@ -49,21 +49,17 @@ let
       version = "5.212.0-alpha4";
     };
 
-    # Even if developed in the public, QtWebEngine does not have official
-    # releases or new tags since the Qt company made 5.15.3 proprietary.
-    # Apparently they care more about licensing than the security of their users.
-    # See https://lists.qt-project.org/pipermail/interest/2021-March/036387.html
     qtwebengine =
       let
-        branchName = "5.15.3";
-        rev = "a059e7404a6db799f4da0ad696e65ae9c854b4b0";
+        branchName = "5.15.5";
+        rev = "v${branchName}-lts";
       in
       {
         version = "${branchName}-${lib.substring 0 7 rev}";
 
         src = fetchgit {
           url = "https://github.com/qt/qtwebengine.git";
-          sha256 = "1vdgxfbmx4z4qrm2g61dl64gqn3fv5f83jwpp7h1gyfx5z2qvfmv";
+          sha256 = "12wf30d34sgn82mbz91xybxyn3j1mhvxda452cfkxm232n1f2kjb";
```

Comment:
For some reason the hash is `1kv483dh72vx84zhc526mc9haaifsdf61shs401qmdnhsyxb0skx` on my machine

Reply:
Let's see if `leaveDotGit = true;` is the problem. With it commented out I get `1ilmaj3mdfafcj74d94k48k5a7rphwpnfch371nkkx74pb42y2fx
`. Do you get the same hash?

Reply:
I can reproduce both `12wf30d34sgn82mbz91xybxyn3j1mhvxda452cfkxm232n1f2kjb` and `1ilmaj3mdfafcj74d94k48k5a7rphwpnfch371nkkx74pb42y2fx` here

Reply:
@prusnak any news? Can you reproduce `1ilmaj3mdfafcj74d94k48k5a7rphwpnfch371nkkx74pb42y2fx` with `leaveDotGit = true;` commented out?

Reply:
I can't reproduce with any of the hashes. That is probably I am using macOS to build. I guess let's ignore this for now and we'll fix the issue later when Qt 5.15 is enabled for darwin (currently 5.12 is used).

Reply:
Most probably the `postFetch` step is not deterministic on darwin.

Reply:
Oh, sure! Thanks for the feedback :)

Reply:
I am wondering - do we really need `leaveDotGit = true`? I am going to try using `fetchFromGithub` with `leaveDotGit = false` and `fetchSubmodules = true`.

Reply:
The following code fetches the sources for me without any issues:

```
    qtwebengine =
      let 
        branchName = "5.15.5";
        rev = "v${branchName}-lts";
      in  
      {   
        version = "${branchName}-${lib.substring 0 7 rev}";

        src = fetchFromGitHub {
          owner = "qt";
          repo = "qtwebengine";
          inherit rev branchName;
          sha256 = "0yyjwjzgl5b8vsbk2z0wg941i5nbif7jb752pbcnbnx8gl6kwga3";
          fetchSubmodules = true;
        };  
      };
```

Can you please try whether this works for you and builds for you?

Reply:
Fetches, but doesn't build:
```
/build/source/src/core/api/qtwebenginecoreglobal_p.h:54:10: fatal error: QtWebEngineCore/qtwebenginecoreglobal.h: No such file or directory
   54 | #include <QtWebEngineCore/qtwebenginecoreglobal.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
There was a reason for doing it like this.

Reply:
I think we should try to fix the issue and not working it around by using fetchgit. But I agree this can be addressed in one of the future pull requests and not this one.